### PR TITLE
Migrate DynamoDB Java API to return CompletionStage instead of Future…

### PR DESCRIPTION
**Pull request by adamtrousdale**
… (#1020)

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -83,6 +83,7 @@ object Dependencies {
 
   val DynamoDB = Seq(
     libraryDependencies ++= Seq(
+      "org.scala-lang.modules" %% "scala-java8-compat" % "0.8.0", // BSD-style
```
Comment (2m) [inline project/Dependencies.scala]: Dependency to java8-compat comes transitively from akka-actor. So it is not needed to include it here as well.

**Comment by adamtrousdale** [comment]
Ah my bad, I missed that. Removed the explicit dependency.

**Comment by ennru** [comment]
Fixes #1020

**Comment by ennru** [comment]
Thank you for getting this done! Keep going.

**Pull request merged**